QuantLib/ql/termstructures/volatility/optionlet/optionletstripper.hpp

```diff
@@ -40,6 +40,7 @@ namespace QuantLib {
       public:
         //! \name StrippedOptionletBase interface
         //@{
+        enum Model{Normal, ShiftedLognormal};
```

Comment:
I vote for {Normal, ShiftedLogNormal}

ShiftedLogNormal conveys the notion that negative rates can be handled, and
if you set the shift=0 then you have the standard lognormal model.

Using just LogNormal would leave the user wondering about negative rates


On Wed, Mar 25, 2015 at 5:59 AM, Michael von den Driesch <
notifications@github.com> wrote:

> In QuantLib/ql/termstructures/volatility/optionlet/optionletstripper.hpp
> <https://github.com/lballabio/quantlib/pull/167#discussion_r27095157>:
>
> > @@ -40,6 +40,7 @@ namespace QuantLib {
> >        public:
> >          //! \name StrippedOptionletBase interface
> >          //@{
> > +        enum Model{Normal, ShiftedLognormal};
>
> I can make the change with regards to the "case naming". But just to be
> sure we have no misunderstanding ...
> *Normal Case*
> [image: equation]
> <https://camo.githubusercontent.com/9fc010d56cc0e0b0638a7f2e3603b6a086b26d15/687474703a2f2f6c617465782e636f6465636f67732e636f6d2f6769662e6c617465783f64462533442535437369676d612532306457>
> *Shifted LogNormal Case*
> [image: equation]
> <https://camo.githubusercontent.com/bf165397780cb0237c1e6fe1c50eded7f957797a/687474703a2f2f6c617465782e636f6465636f67732e636f6d2f6769662e6c617465783f64253238462532302b253230612532392533446446253344253238462b612532392535437369676d612532306457>
>
> where *a* is the shift (i.e. displacement within the constructor). So
> there is no shift in the normal case whereas the shifted LogNormal case
> contains the usual LogNormal case with *a=0* as a special case. This has
> been the reason for my naming convention.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lballabio/quantlib/pull/167/files#r27095157>.
>